RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -124,246 +157,804 @@ void L2MuonSeedGeneratorFromL1T::produce(edm::Event& iEvent, const edm::EventSet
+            for(j=0; j<NofflineSeed1; ++j) {
+              dRmtx[theL1][j] = 999;
+              dRmtx[i][theOffs] = 999;
+            }
+          }
+
+          if( selOffseeds[theL1][theOffs] != nullptr ) {
+            //put given L1mu and offSeed to output
+            edm::OwnVector<TrackingRecHit> Newcontainer;
+
+            PTrajectoryStateOnDet const & seedTSOS = selOffseeds[theL1][theOffs] ->startingState();
+                          TrajectorySeed::const_iterator
+                          tsci  = selOffseeds[theL1][theOffs]->recHits().first,
+                          tscie = selOffseeds[theL1][theOffs]->recHits().second;
+                          for(; tsci!=tscie; ++tsci) {
+                            Newcontainer.push_back(*tsci);
+                          }
+                          output->push_back(L2MuonTrajectorySeed(seedTSOS,Newcontainer,alongMomentum,
+                                            MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),Newit)  )));
+          }
+        }
+      }
+
+      else if(matchType==4) {
+
+        for(i=0; i<NmuColl; ++i) {
+
+          auto Newit = muColl->begin(muColl->getFirstBX()) + i;
+
+          double tempDR = 999;
+          int theOffs = 0;
+
+          double NewdRcone = matchingDR[0];
+          if ( fabs(Newit->eta()) < etaBins.back() ){
+            std::vector<double>::iterator lowEdge = std::upper_bound (etaBins.begin(), etaBins.end(), fabs(Newit->eta()));
+            NewdRcone    =  matchingDR.at( lowEdge - etaBins.begin() - 1);
+          }
+
+          for(j=0; j<NofflineSeed1; ++j) {
+            if( tempDR > dRmtx[i][j] ) {
+              tempDR = dRmtx[i][j];
+              theOffs = j;
+            }
+          }
+
+          if( !(tempDR < NewdRcone) )  continue;
+
+          if( selOffseeds[i][theOffs] != nullptr ) {
+            //put given L1mu and offSeed to output
+            edm::OwnVector<TrackingRecHit> Newcontainer;
+
+            PTrajectoryStateOnDet const & seedTSOS = selOffseeds[i][theOffs] ->startingState();
+                          TrajectorySeed::const_iterator
+                          tsci  = selOffseeds[i][theOffs]->recHits().first,
+                          tscie = selOffseeds[i][theOffs]->recHits().second;
+                          for(; tsci!=tscie; ++tsci) {
+                            Newcontainer.push_back(*tsci);
+                          }
+                          output->push_back(L2MuonTrajectorySeed(seedTSOS,Newcontainer,alongMomentum,
+                                            MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),Newit)  )));
+          }
+        }
+      }
+
+    } // useOfflineSeed && isAsso
+
+  } // matchType > 0
+
+  //--- SortType 1 : by L1 pT
+  if(sortType == 1) {
+    sort(output->begin(),output->end(),SortByL1Pt);
+  }
+
+  //--- SortType 2 : by L1 quality and pT
+  else if(sortType == 2) {
+    sort(output->begin(),output->end(),SortByL1QandPt);
+  }
+
+  /*
```

Comment:
Please remove the commented out code that you included for your debug (or use some LOgDebug if you want to maintain here for possible future needs)